packages/clay-multi-select/src/MultiSelect.tsx

```diff
@@ -0,0 +1,196 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import React from 'react';
+
+const COMMA = 188;
+
+const KEYS = [COMMA];
+
+export interface IProps extends React.HTMLAttributes<HTMLInputElement> {
+	children?: React.ReactChild | React.ReactChild[];
+	errorMessage?: string;
+	forwardRef?: React.Ref<HTMLDivElement>;
+	helpText?: string;
+	inputValue?: string;
+	items: string[];
+	label?: React.ReactText;
+	onItemsChange: (val: string[]) => void;
+	onInputChange?: (val: string) => void;
+	spritemap?: string;
+	validateOnBlur?: boolean;
+	validationFn?: (val?: string) => boolean;
+}
+
+const MultiSelect: React.FunctionComponent<IProps> = ({
```

Comment:
I've missed some interactions in this component, it's probably something we should deliver on a high-level component.

- Press backspace in the input when it does not have a value, it takes to the focus of the Label, pressing again, erases the label.
- Navigate the label with Tab or keyboard
- When deleting an item, you must return the focus to the input...

This is the technical document with use cases and interactions: https://docs.google.com/document/d/1cIx6RkWNbnho_fOSR__3Fp5alIBE_ObhO-jecfXlA9I/edit#heading=h.jkrp0t7cj15n

Reply:
For future reference, we should ping @drakonux (and open review requests) as we implement the components.

Also, maybe we should start committing the specification documents so we can keep track of them here and have them versioned... wdt?

Reply:
Do we have docs like this for all clay components?

Reply:
Yeah, we most certainly do, we just need to ask for them and maybe keep them closer to heart.

This is something we've often discussed regarding our workflow. Documents in google docs are live and subject to change without notice. This has resulted in doubts as to wether a component was wrongly implemented or if the definition changed underneath. If we could keep the definitions here and version them through git, that might alleviate some of the pain and also help collect our requirements.

What do you think?

Reply:
I would think it would be great, keep the documents in git and versioned, we need to versioned these documents to not get some of the problems we had in the 2.x life, I got a lot of complaints because Clay was "inconsistent" with Lexicon (this was a problem when we adopt monolithic components...). But do you say keep our own documents or in conjunction with Lexicon? One good thing is that we can collaborate much more closely with them if we keep those components together.